Enable syncing of windows packages

raskchanky opened the pull request:
There is also a companion PR that updates the builder depot client, https://github.com/habitat-sh/habitat/pull/5154

![](https://media.giphy.com/media/6uGhT1O4sxpi8/giphy.gif)

Signed-off-by: Josh Black <raskchanky@gmail.com>

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

chefsalim commented on Cargo.lock:
```diff
@@ -1129,19 +1129,18 @@ dependencies = [
  "hyper 0.10.13 (registry+https://github.com/rust-lang/crates.io-index)",
- "hyper-openssl 0.5.0 (registry+https://github.com/rust-lang/crates.io-index)",
- "log 0.4.1 (registry+https://github.com/rust-lang/crates.io-index)",
+ "hyper-openssl 0.2.7 (registry+https://github.com/rust-lang/crates.io-index)",
```
Hmm are all these lock file changes intentional?

raskchanky replied:
No, this was a result of patching the `Cargo.toml` to pull the `habitat` crate in by path.  I'll revert.

chefsalim commented on components/builder-depot/src/upstream.rs:
```diff
@@ -163,36 +168,40 @@ impl UpstreamMgr {
+                let target = upstream_request.get_target();
 
-                // We only care about the base ident
-                let ident =
```
The main thing with moving this to check_request is that we could have many different variants of the same package (eg, if the user was browsing a list of versions of a package in the UI) that will trigger N different calls to the upstream.  Though it might be ok since we are discussing switching over to fetching a specific package instead of just the latest..

chefsalim commented:
Looks good overall.. just a couple of comments.

raskchanky commented:
@chefsalim I've addressed your comments, though I don't think Travis will pass on this PR until https://github.com/habitat-sh/habitat/pull/5154 is merged
